spec/support/request/tomselect_helper.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module TomselectHelper
+  def tomselect_open(field_name)
+    page.find("##{field_name}-ts-control").click
+  end
+
+  def tomselect_multiselect(value, options)
+    tomselect_wrapper = page.find_field(options[:from]).sibling(".ts-wrapper")
+    tomselect_wrapper.find(".ts-control").click
+    tomselect_wrapper.find(:css, '.ts-dropdown.multi .ts-dropdown-content .option',
+                           text: value).click
+  end
+
+  def tomselect_search_and_select(value, options)
+    tomselect_wrapper = page.find_field(options[:from]).sibling(".ts-wrapper")
+    tomselect_wrapper.find(".ts-control").click
+    # Use send_keys as setting the value directly doesn't trigger the search
+    tomselect_wrapper.find(:css, '.ts-dropdown input.dropdown-input').send_keys(value)
+    tomselect_wrapper.find(:css, '.ts-dropdown .ts-dropdown-content .option', text: value).click
+  end
+
+  def tomselect_select(value, options)
+    tomselect_wrapper = page.find_field(options[:from]).sibling(".ts-wrapper")
+    tomselect_wrapper.find(".ts-control").click
+
+    tomselect_wrapper.find(:css, '.ts-dropdown .ts-dropdown-content .option', text: value).click
+  end
+
+  def open_tomselect_to_validate!(page, field_name)
```

Comment:
Why the "!" in the name here ? the function is not changing anything ?

Reply:
Sometimes `!` is used for methods that raise an exception for the normal fail pathway. (eg `save` returns `true/false`, `save!` returns `true` or raises exception).
Hmm, but the exception is not a normal fail pathway, so you're right, `!` is unnecessary here.

Reply:
Yes, David exactly.

Moreover David, Gaetan - I think it makes sense to use it with method names which have intentional code breaking statements like exceptions, or expect (_even though it doesn't raise the exception but it does break the code flow by not executing further lines)_

Like if the `expect` gets failed, then the spec in which this method is being used, won't be continuing further. 😅

What do you think about it? Thanks

Reply:
I think it doesn't matter much either way ;)